ddtrace/tracer/metrics.go

```diff
@@ -93,8 +93,22 @@ func (t *tracer) reportHealthMetricsAtInterval(interval time.Duration) {
 	for {
 		select {
 		case <-ticker.C:
-			t.statsd.Count("datadog.tracer.spans_started", int64(atomic.SwapUint32(&t.spansStarted, 0)), nil, 1)
-			t.statsd.Count("datadog.tracer.spans_finished", int64(atomic.SwapUint32(&t.spansFinished, 0)), nil, 1)
+			// if there are started spans, report the number spans with their integration, then
+			// reset the count
+			t.spansStarted.Range(func(key string, value *atomic.Int64) bool {
+				err := t.statsd.Count("datadog.tracer.spans_started", value.Load(), []string{"integration:" + key}, 1)
+				t.spansStarted.Delete(key)
+				return err == nil
```

Comment:
https://pkg.go.dev/github.com/puzpuzpuz/xsync/v3#MapOf.Range
> If f returns false, range stops the iteration

If I'm understanding this correctly, I don't think we want this behavior; if Count fails on one occasion, it could still succeed on future occasions. So we'd want to attempt to send every single entry, even if the previous one fails. Instead of returning whether the error was nil, we could:

- if err not nil, log about the dropped data
- always return true in anonymous function (we don't want a stop condition)

See: https://pkg.go.dev/github.com/datadog/datadog-go/statsd#Client.Count